src/evict/evict_lru.c

```diff
@@ -1256,6 +1295,14 @@ __evict_push_candidate(WT_SESSION_IMPL *session,
 	evict->btree = S2BT(session);
 	evict->ref = ref;
 	evict->score = __evict_read_gen(evict);
+
+	/* Adjust for size when doing dirty eviction. */
+	if (FLD_ISSET(S2C(session)->cache->state, WT_EVICT_STATE_DIRTY) &&
+	    evict->score != WT_READGEN_OLDEST && evict->score != UINT64_MAX &&
+	    !__wt_page_is_modified(ref->page))
+		evict->score += WT_MEGABYTE -
+		    WT_MIN(WT_MEGABYTE, ref->page->memory_footprint);
```

Comment:
I don't know how useful this really is: the intent is to make larger dirty pages more attractive eviction candidates when we're trying to reduce the dirty %.  Since this branch also fixes some bugs with the urgent queue, this may no longer be worthwhile.